gateway-js/src/index.ts

```diff
@@ -467,6 +445,102 @@ export class ApolloGateway implements GraphQLService {
     }
   }
 
+  private async updateCsdl(result: CsdlUpdate): Promise<void> {
```

Comment:
There's a lot of copy and paste between this and updateServiceDefs, though there's a fair amount of different stuff too. How hard would it be to factor more of that out?  Or are we hoping to be able to permanently delete updateServiceDefs sooner rather than later?

Reply:
I tried a few things, and spent maybe longer than I should have trying to come up with something I liked better and I just didn't, but I get the desire to dry this up.

Sadly, we won't be getting rid of this branch in the code as long as we allow users to provide their own update function which returns the old (current) format.

This may play well into your suggestion to create a new hook and deprecate the old one. If we do that, we make composition the responsibility of the customer in their new update function which returns csdl (not hard to do).